src/main/java/edu/cornell/kfs/fp/service/impl/AmazonWebServicesBillingServiceImpl.java

```diff
@@ -0,0 +1,649 @@
+        return results;
+    }
+    
+    protected ClientRequest buildClientRequest() {
+        ClientRequest.Builder builder = new ClientRequest.Builder();
+        builder.accept(MediaType.APPLICATION_JSON);
+        builder.header(CuFPConstants.AmazonWebServiceBillingConstants.AUTHORIZATION_HEADER_NAME, 
+                CuFPConstants.AmazonWebServiceBillingConstants.AUTHORIZATION_TOKEN_VALUE_STARTER + getAwsToken());
+        URI uri;
+        try {
+            uri = new URI(getAwsURL() + CuFPConstants.AmazonWebServiceBillingConstants.URL_PARAMETER_YEAR + findProcessYear() + 
+                    CuFPConstants.AmazonWebServiceBillingConstants.URL_PARAMETER_MONTH + findProcessMonthNumber());
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+        ClientRequest request = builder.build(uri, CuFPConstants.AmazonWebServiceBillingConstants.HTTP_METHOD_GET_NAME);
+        return request;
+    }
+    
+    protected List<AmazonAccountDetail> buildAmazonAcountListFromJson(String jsonResults) {
+        ObjectMapper mapper = new ObjectMapper();
+        AmazonAccountDetailContainer detailContainer = new AmazonAccountDetailContainer();
+        try {
+            detailContainer = mapper.readValue(jsonResults, AmazonAccountDetailContainer.class);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return detailContainer.getAccountDetail();
+    }
+
+    protected boolean processAmazonAccountDetails(List<AmazonAccountDetail> amazonAccountDetails) {
+        boolean valid = true;
+        int count = 0;
+        for (AmazonAccountDetail amazonAccountDetail : amazonAccountDetails) {
+            if (StringUtils.equalsIgnoreCase(CuFPConstants.AmazonWebServiceBillingConstants.INTERNAL_KFS_ACCOUNT_DESCRIPTION, amazonAccountDetail.getKfsAccount())) {
+                LOG.info("processAmazonAccountDetails() Found an internal kfs account, ignoring this transaction for AWS account " 
+                    + amazonAccountDetail.getAwsAccount());
+            } else {
+                valid = processExternalBill(amazonAccountDetail) && valid;
+            }
+            count ++;
+        }
+        LOG.info("Processed " + count + " AWS records");
+        return valid;
+    }
+
+    private boolean processExternalBill(AmazonAccountDetail amazonAccountDetail) {
+        boolean valid = true;
+        try {
+            TransactionDTO transactionDTO = buildTransactionDTO(amazonAccountDetail);
+            if (transactionDTO.isFailingError()) {
+                valid = false;
+                LOG.error("processAmazonAccountDetails() there errors that prevent a creation of a DI " + amazonAccountDetail.toString());
+            } else if (shouldBuildDistributionOfIncomeDocument(amazonAccountDetail, transactionDTO.getAmount())) {
+                buidAndRouteDistributionOfIncomeDocument(amazonAccountDetail, transactionDTO);
+            }
+        } catch (Throwable e) {
```

Comment:
I wanted to make sure that any error happening in be transaction doesn't prevent subsequent ones